test/preview/components/retrievers/test_memory_retriever.py

```diff
@@ -0,0 +1,125 @@
+
+    @pytest.mark.unit
+    def test_invalid_run_wrong_store_name(self):
+        # Test invalid run with wrong store name
+        ds = MemoryDocumentStore()
+        mr = MemoryRetriever(document_store_name="memory")
+        with pytest.raises(ValueError, match=r"MemoryRetriever's document store 'memory' not found"):
+            invalid_input_data = MemoryRetriever.Input(
+                query="test", top_k=10, scale_score=True, stores={"invalid_store": ds}
+            )
+            mr.run(invalid_input_data)
+
+        with pytest.raises(ValueError, match=r"MemoryRetriever can only be used with a MemoryDocumentStore."):
+            invalid_input_data = MemoryRetriever.Input(
+                query="test", top_k=10, scale_score=True, stores={"memory": "not a MemoryDocumentStore"}
+            )
+            mr.run(invalid_input_data)
+
+    @pytest.mark.unit
+    def test_invalid_run_wrong_store_type(self):
+        # Test invalid run with wrong store type
+        ds = MemoryDocumentStore()
+        mr = MemoryRetriever(document_store_name="memory")
+        with pytest.raises(ValueError, match=r"MemoryRetriever can only be used with a MemoryDocumentStore instance."):
+            invalid_input_data = MemoryRetriever.Input(
+                query="test", top_k=10, scale_score=True, stores={"memory": "not a MemoryDocumentStore"}
+            )
+            mr.run(invalid_input_data)
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize(
+        "query, query_result",
+        [
+            ("Javascript", "Javascript is a popular programming language"),
+            ("Java", "Java is a popular programming language"),
+        ],
+    )
+    def test_run_with_pipeline(
+        self, test_memory_retriever_docs: List[Document], query: str, query_result: str, top_k: int = 1
+    ):
+        self.pipeline_run(test_memory_retriever_docs, query=query, query_result=query_result, top_k=top_k)
+
+    @pytest.mark.integration
+    def test_run_with_pipeline_and_top_k(self, test_memory_retriever_docs: List[Document]):
+        self.pipeline_run(
+            test_memory_retriever_docs, query="Python", query_result="Python is a popular programming language", top_k=3
+        )
+
+    def pipeline_run(self, test_memory_retriever_docs: List[Document], query: str, query_result: str, top_k: int):
+        ds = MemoryDocumentStore()
+        ds.write_documents(test_memory_retriever_docs)
+        mr = MemoryRetriever(document_store_name="memory")
+
+        pipeline = Pipeline()
+        pipeline.add_component("retriever", mr)
+        pipeline.add_store("memory", ds)
+        result: Dict[str, Any] = pipeline.run(data={"retriever": MemoryRetriever.Input(query=query, top_k=top_k)})
+
+        assert result
+        assert "retriever" in result
+        results_docs = result["retriever"].documents
+        assert results_docs
+        assert len(results_docs) == top_k
+        assert results_docs[0].content == query_result
```

Comment:
Not a bad idea in general, but I would just duplicate the body into the two tests. This method takes many parameters and doesn't really save us many lines of code. We can rethink this later if we see this pattern repeats more often.